contracts/Bucket.sol

```diff
@@ -1,15 +1,91 @@
 pragma solidity ^0.6.1;
 pragma experimental ABIEncoderV2;
 
-library RedeemUtil {
+abstract contract Bucket {
+  bool initialized;
+  address payable public owner;
+  address public tokenAddress;
+  uint256 public expirationTime;
+  uint256 public startTime;
+
+  uint256 constant maxTxDelayInBlocks = 10;
+  bytes32 constant REDEEM_TYPEHASH = keccak256("Redeem(uint256 blockNumber,bytes32 blockHash,address receiver,bytes32 code)");
+  bytes32 constant EIP712DOMAIN_TYPEHASH = keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
+  bytes32 DOMAIN_SEPARATOR;
+
+  struct Gift {
+    address recipient;
+    bytes32 code;
```

Comment:
can we leave this uint256?

Reply:
the redeem code has always been bytes32, would you prefer to change it to uint256?

Reply:
sorry my error :D